cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/TriggerCodec.java

```diff
@@ -34,9 +38,21 @@
       .put(ProtoTrigger.Type.PARTITION, PartitionTrigger.class)
       .put(ProtoTrigger.Type.STREAM_SIZE, StreamSizeTrigger.class)
       .put(ProtoTrigger.Type.PROGRAM_STATUS, ProgramStatusTrigger.class)
+      .put(ProtoTrigger.Type.AND, AndTrigger.class)
+      .put(ProtoTrigger.Type.OR, OrTrigger.class)
       .build();
 
   public TriggerCodec() {
     super(TYPE_TO_INTERNAL_TRIGGER);
   }
+
+  @Override
+  public Trigger deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
+    throws JsonParseException {
+    Trigger trigger = super.deserialize(json, typeOfT, context);
+    if (trigger instanceof AbstractCompositeTrigger) {
+      ((AbstractCompositeTrigger) trigger).initializeUnitTriggers();
```

Comment:
Would it be better to make `getUnitTriggers()` initialize the unit triggers if they haven't been initialized already?